File: c16/c16.php
<?php
//ini_set('memory_limit', '512M');
$input=explode(",",readline());

//No pude resistirme, pero BASH es lento para el resto del proceso

shell_exec("sed -n ".$input[0].",".($input[0]+$input[1]-1)."p c16/points | tr  '\011' ',' | tr -d ' ' > subset");

$objects=array();
$handle =fopen("subset",'r');

while (1)
{
	$line=fgets($handle);
	if ($line=="") break;
	$cosas=explode(",",$line);
	$objects[]=array($cosas[0], $cosas[1], $cosas[2], false);
}
echo "Parseo ok\n";

$count=0;
$c=0;
for ($i=0;$i<$input[1];$i++)
{
	//echo $c."\n";
	for ($j=0;$j<$input[1];$j++)
	{
		
		if ($objects[$j][3] == true)
			continue;
		//echo "------------------------\n";
		//echo "(".$objects[$i][0]."-".$objects[$j][0].")^2=".pow($objects[$i][0]-$objects[$j][0],2)."\n";
		//echo "(".$objects[$i][1]."-".$objects[$j][1].")^2=".pow($objects[$i][1]-$objects[$j][1],2)."\n";
		//echo "Total: ".(pow($objects[$i][0]-$objects[$j][0],2)+pow($objects[$i][1]-$objects[$j][1],2))."\n";
		//echo "(".$objects[$i][2]."-".$objects[$j][2].")^2=".pow($objects[$i][2]-$objects[$j][2],2)."\n";
		
		//echo "------------------------\n";
		
		if ($objects[$i][0] == $objects[$j][0] && $objects[$i][1] == $objects[$j][1])
			continue;
		
		if ( pow($objects[$i][0]-$objects[$j][0],2)+pow($objects[$i][1]-$objects[$j][1],2) <= pow($objects[$i][2]+$objects[$j][2],2))
				$count++;
	
	}
	$objects[$i][3]=true;
	$c++;

}
echo $count."\n";
?>

File: c16/c16.c
#include <stdio.h>
#include <stdlib.h>

int main ()
{
	unsigned long start, lines;
	scanf("%lu,%lu",&start, &lines);
	
	char cmd[200];
	sprintf(cmd, "sed -n %lu,%lup c16/points | tr  '\\011' ',' | tr -d ' ' > subset", start, start+lines);
	system(cmd);
	
	FILE * fp;
	char * line = NULL;
	size_t len = 0;
	ssize_t read;

	fp = fopen("subset", "r");
	
	typedef struct Object{
		unsigned long x;
		unsigned long y;
		unsigned long r;
	} Object;
	
	Object objects [lines+1];

	unsigned long count=0;
	while ((read = getline(&line, &len, fp)) != -1) {
		sscanf("%lu,%lu,%lu",line, &(objects[count].x),&(objects[count].y),&(objects[count].r));
		count++;
		
	}
	if (line)
	free(line);
	unsigned long i;
	for (i=0;i<count;i++)
		printf("X:%lu Y:%lu R:%lu\n",objects[i].x,objects[i].y,objects[i].r);

	
	exit(EXIT_SUCCESS);
	
}
/*
$input=explode(",",readline());

//No pude resistirme, pero BASH es lento para el resto del proceso
shell_exec("sed -n ".$input[0].",".($input[0]+$input[1])."p c16/points | tr  '\011' ',' | tr -d ' ' > subset");

$objects=array();
$handle =fopen("subset",'r');

while (1)
{
	$line=fgets($handle);
	if ($line=="") break;
	$cosas=explode(",",$line);
	$objects[]=$cosas;
}
//echo "Parseo ok\n";

$count=0;
$c=0;
foreach ($objects as $c1)
{
	//echo $c."\n";
	foreach($objects as $c2)
	{
		if ( pow($c1[0]-$c2[0],2)+pow($c1[1]-$c2[1],2) < pow($c1[2]-$c2[2],2) 
			&& pow($c1[0]-$c2[0],2)+pow($c1[1]-$c2[1],2) != 0 )
			$count++;
	
	}
	$c++;

}
echo $count."\n";
?>*/